core/src/main/java/org/elasticsearch/discovery/DiscoveryService.java

```diff
@@ -132,7 +133,7 @@ public void publish(ClusterChangedEvent clusterChangedEvent, Discovery.AckListen
     }
 
     public static String generateNodeId(Settings settings) {
-        Random random = Randomness.get(settings, DiscoveryService.SETTING_DISCOVERY_SEED);
+        Random random = Randomness.get(settings, DiscoveryService.DISCOVERY_SEED_SETTING.getKey());
```

Comment:
Would it make sense to create a new method in `Randomness` with the following signature: `public static Random get(Settings settings, Setting<Long> setting)`? Now we "unpack" the key, just to get the settings the old way again.